### ISSUE TITLE
Discrepancy Between Reported and Reproduced Results

### ISSUE THREAD

**Issue by SKLols** [kumuji/stu_dataset#3]
Hi, thank you for sharing this excellent work and the accompanying code. I’m currently reproducing the results from your paper and wanted to share my approach and observations to request your feedback and clarify a few points. I attempted to reproduce the results for Deep Ensemble and MC Dropout method with the help of checkpoints as presented in the paper.

**My Results**
**Deep Ensemble:**
I ran inference using 3 ensemble checkpoints with the max_logit function given in pq_trainer.py 

`def get_maxlogit(logit, mask, inv_map):
    confid = mask.float().sigmoid().matmul(logit)
    max_logit = torch.max(confid, dim=1).values[inv_map]
    max_logit = (max_logit * -1) + 1
    return max_logit
`
I got Individual result as 
- Ensemble_0: AUROC 89.95 | FPR@95 65.79 | AP 3.684
- Ensemble_1: AUROC 90.65 | FPR@95 46.72 | AP 6.2637
- Ensemble_2: AUROC 87.37 | FPR@95 66.68 | AP 1.9783
- Ensemble_avg: AUROC 89.32 | FPR@95 59.73 | AP 3.98
I compared this average with paper. This average result is noticeably higher than the paper’s AUROC (86.74) and slightly off in other metrics.

**MC Dropout:**
Used the MC Dropout checkpoint with max_logit criteria as well
Result: AUROC 83.78 | FPR@95 98.29 | AP 0.41

**My Questions**

1. Is it standard practice to compute the average of the ensemble results for AUROC, FPR@95, and AP as I did above?
2. For MC Dropout and Void Classifier, is it correct to use the max_logit function when saving predictions during inference?
3. For the Max Logit baseline mentioned in the paper (AUROC 84.53), is it sufficient to just train the model and apply the max_logit function during inference to obtain this result?

I'd greatly appreciate your insights on whether my reproduction method aligns with your original setup, and if there are specific steps or scripts I might have misunderstood.
Thanks again for this valuable contribution to the community.

Best Regards,
Sourabh

**Comment by kumuji** [comment]
> I trained the model from scratch using the same settings you provided. The only change I made was reducing the batch size from 16 to 4 (due to hardware constraints on my RTX 4090). All other parameters, including the configuration and training scripts, remained unchanged. The seed was set to None as in the original config.

Although we haven't tried training the model with these parameters, in principle it should work in a similar way and have the same performance for the same number of steps.

> After training, I evaluated the checkpoint with the best PQ score and got the following results:
epoch_26_pq_0.616 AUROC : 82.86;  FPR@95 : 98.46;  AP : 1.706;  Threshold: -0.0125
The results, especially the FPR, differ significantly from those of any of the ensemble models or Max Logit reported in the paper or in your earlier message. I wanted to ask if this kind of variation is expected due to:

Setting the seed to random is fine. If you train using all the datasets (semkitti+panopticcudal+stu_train), the numbers are simply due to the batch size. However, looking at the numbers, it seems that the model performs very well on the SemKitti validation set, but not so well on the anomaly segmentation task. Please try testing your model on the Stu `201` scene to see if you can achieve good inlier segmentation performance.

> Also, as you kindly offered earlier, I would really appreciate it if you could push the implementation of the Void Classifier and MC Dropout methods to a separate branch whenever convenient. I'd love to experiment further with these baselines for comparison.

I had intended to write the reply after publishing the void and mcdrop code, and had hoped to do so within the past two weeks. Unfortunately, this time of year is always very busy, so I will have to come back to this a bit later. Sorry.

**Comment by SKLols** [comment]
Hi Alexey,
Thanks for your suggestion I tried testing our model on Stu 201 scene as you recommended.

Could you please clarify the difference between Max Logit and Deep Ensemble?

My understanding:
1) Max Logit = single model + max_logit function.
2) Deep Ensemble = multiple independently trained models combined via entropy. (In this case 3 models)

Is my understanding correct?   
If it is correct? If so, does that mean a single trained checkpoint can act both as a “Max Logit” model (if used alone) and as one “ensemble member” (if combined with others)?

Also, a quick clarification regarding the confid calculation in max_logit and RbA functions: I noticed that logit values are passed through softmax before computing confid. Does this mean Max Logit is computed using maximum softmax probability (MSP) rather than raw logits? I just want to make sure I understand the procedure correctly.

Thanks,
Sourabh

**Comment by chrisyan** [comment]
Hi Alexey,
thank you very much for the amazing work! Could you upload code for the void classifier and MC dropout? 
Thanks!
